agent/functions/put_file_test.go

```diff
@@ -0,0 +1,193 @@
+				}
+				if got.Path != "existing.txt" || got.Content != "new content\n" {
+					t.Errorf("unexpected result: got %+v", got)
+				}
+			},
+		},
+		{
+			name: "異常系：無効なパス（ファイル名なし）",
+			input: PutFileInput{
+				OutputPath:  "/invalid/path/",
+				ContentText: "test content",
+			},
+			wantErr: true,
+			validateFunc: func(t *testing.T, got store.File, err error) {
+				if err == nil {
+					t.Error("expected error but got nil")
+				}
+			},
+		},
+		{
+			name: "異常系：書き込み権限なしディレクトリ",
+			input: PutFileInput{
+				OutputPath:  "noperm/test.txt",
+				ContentText: "test content",
+			},
+			setupFunc: func(t *testing.T, dir string) {
+				noPermDir := filepath.Join(dir, "noperm")
+				if err := os.Mkdir(noPermDir, 0000); err != nil {
+					t.Fatal(err)
+				}
+			},
+			wantErr: true,
+			validateFunc: func(t *testing.T, got store.File, err error) {
+				if err == nil {
+					t.Error("expected error but got nil")
+				}
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// テスト用の一時ディレクトリを作成
+			tmpDir := t.TempDir()
+			
+			// 現在のディレクトリを保存
+			currentDir, err := os.Getwd()
+			if err != nil {
+				t.Fatal(err)
+			}
+			
+			// テストディレクトリに移動
+			if err := os.Chdir(tmpDir); err != nil {
+				t.Fatal(err)
+			}
+			
+			// テスト終了時に元のディレクトリに戻る
+			t.Cleanup(func() {
+				os.Chdir(currentDir)
+			})
+
+			// セットアップ関数の実行
+			if tt.setupFunc != nil {
+				tt.setupFunc(t, tmpDir)
+			}
+
+			// テスト実行
+			got, err := PutFile(tt.input)
+
+			// バリデーション関数の実行
+			if tt.validateFunc != nil {
+				tt.validateFunc(t, got, err)
+			}
+
+			// ファイルの内容を確認
+			if tt.checkFile && !tt.wantErr {
+				content, err := os.ReadFile(tt.input.OutputPath)
+				if err != nil {
+					t.Errorf("failed to read file: %v", err)
+				}
+				expectedContent := tt.input.ContentText
+				if expectedContent[len(expectedContent)-1] != '\n' {
+					expectedContent += "\n"
+				}
+				if string(content) != expectedContent {
+					t.Errorf("file content mismatch: got %q, want %q", string(content), expectedContent)
+				}
+			}
+		})
+	}
+}
+
+func TestInitPutFileFunction(t *testing.T) {
+	f := InitPutFileFunction()
+	
+	if f.Name != FuncPutFile {
+		t.Errorf("unexpected function name: got %s, want %s", f.Name, FuncPutFile)
+	}
+	
+	if f.Description != "Put new content to the file" {
+		t.Errorf("unexpected description: got %s", f.Description)
+	}
+	
+	if f.Func == nil {
+		t.Error("function should not be nil")
+	}
+	
+	// パラメータの検証
+	params, ok := f.Parameters["properties"].(map[string]interface{})
+	if !ok {
+		t.Error("properties should be a map")
+		return
+	}
+	
+	// 必須パラメータの検証
+	required, ok := f.Parameters["required"].([]string)
+	if !ok {
+		t.Error("required should be a string slice")
+		return
+	}
+	
+	expectedRequired := []string{"output_path", "content_text"}
+	if len(required) != len(expectedRequired) {
+		t.Errorf("unexpected required parameters: got %v, want %v", required, expectedRequired)
+	}
+}
```

Comment:
from security_reviewer
パストラバーサル攻撃に対するテストケースが不足しています。'../'や'..\' を含むパスなど、ディレクトリトラバーサルの試行に対するテストを追加する必要があります。

```suggestion
以下のようなテストケースを追加することを推奨します：
```go
{
    name: "異常系：ディレクトリトラバーサル攻撃",
    input: PutFileInput{
        OutputPath:  "../../../etc/passwd",
        ContentText: "malicious content",
    },
    wantErr: true,
},
```
```